### ISSUE TITLE
Beast Training displayed costs differ from actual costs (Hunter spell)

### ISSUE THREAD

**Issue by mhdemn** [Atlantiss/NetherwingBugtracker#1900]
![image](https://user-images.githubusercontent.com/25901582/47594575-43b66880-d97c-11e8-9de1-343b2597efb1.png)
Skill: https://tbc-twinhead.twinstar.cz/?spell=5149

**Description**: Hunter's Beast Training displays a training point cost for the different skills you purchase/train your pet that differs from what they actually cost.

**Current behaviour**: Great Stamina Rank 1 says it costs 5 and it does, but the next rank should take 5 from you yet it takes only 5, same with the next rank, can't remember how many do the ranks 4 and 5 cost but definitely half or less of what they should cost. Same thing with Natural Armor. I think I had 80 training points and I could learn Cobra Reflexes (15 training points), Great Stamina ranks 1-5 (1, 5, 10, 15, 25 and 50 training points) and Natural Armor ranks 1-4 (1, 5, 10 and 15 training points).

**Expected behaviour**: Same displayed and actual cost of training points. 

**Server Revision**: 2214

**Comment by DeltaLoads** [comment]
did you deduct the number of points that are required for skill that pet has already learned? for example if pet has already learned rank 4, for rank 5 it deducts points required by amount it takes to learn rank 4

**Comment by mhdemn** [comment]
> did you deduct the number of points that are required for skill that pet has already learned? for example if pet has already learned rank 4, for rank 5 it deducts points required by amount it takes to learn rank 4

I thought about that but I think someone should look into a it 'cause if that was the case, after learning the 4 ranks of Natural Armor I learned I should have been left with a number of training points that would not be a multiple of 5, and the last skill I purchased left me with a round 0 points, I think it was Stamina rank 5 that indicated 50 points to learn and actually displayed the number in a red font as WoW does when you can't afford / aren't eligible to buy something, and I only had 25 (iirc) and It let me purchase them.

**Comment by mhdemn** [comment]
I got a few more training points with a new pet today, a wind serpent, and I couldn't reproduce the bug. It only happened with the standard tiger pet you get with a default 40 hunter. It works as DeltaLoads suggests, each rank deducts the cost of the previous one yet it doesn't display the new cost, so the cost a rank displays is not the cost of that rank but the accumulated cost of that and every previous rank of that same skill.

**Comment by Easyfraggy** [comment]
https://github.com/Atlantiss/NetherwingBugtracker/issues/610

**Comment by Easyfraggy** [comment]
This post has more info, so unlike normally I'll close the original post. Please do use the search function before posting bug reports.

**Comment by mhdemn** [comment]
My apologies about that.
Furthermore, I should add it's likely that Blizzard let this be in retail, as this addon I got recently does the math for you and works in a way that makes me think one of its main features is solving that small issue for the player. About that Natural Armor rank 1 that seems it costed 5 instead of 1 I'm inclined to think it's a bug related to the template pet you guys included in the 40 hunter template rather than a common bug with all pets, since I've tried to reproduce it again with a new pet and it costed 1 again, as it should.

**Comment by mhdemn** [comment]
Just a quick update:

1. Haven't been able to reproduce the cost bug in Natural Armor rank 1. I'm assuming it was a one time thing / brain bug from my side / a thing with the template pet for the default 40 hunter.

2. About the matter of discerning if how Blizzard displayed costs in the pet training window, I found reliable proof it is correct in NW (in case I made anyone actually doubt it was correct, I'm sorry about that)

This is how it looks in Netherwing: 
![image](https://user-images.githubusercontent.com/25901582/47733486-9a67bf00-dc68-11e8-946c-f09ac4b7327b.png)

This is how it looked in retail TBC:
![image](https://user-images.githubusercontent.com/25901582/47733639-ec104980-dc68-11e8-9cb3-4632b76301b7.png)
Source: http://web.archive.org/web/20080619084812/http://petopia.brashendeavors.net:80/html/articles/skills_main.php
saved copy from the petopia website, just how it looked in 2008 (before 3.x.x)

So yeah, once I get a few training points with my next pet / next two pets I'll make sure the Natural Armor bug was an isolated case. But this post ended up being super big for such a small thing so whether you guys wanna close it or not is up to you, I think we're good with everything presented here.
